core/src/main/java/org/infinispan/factories/EncoderRegistryFactory.java

```diff
@@ -64,6 +65,7 @@ public Object construct(String componentName) {
       encoderRegistry.registerTranscoder(new ProtostreamTranscoder(ctxRegistry, classLoader));
       encoderRegistry.registerTranscoder(new JavaSerializationTranscoder(classWhiteList));
       // Wraps the GlobalMarshaller so that it can be used as a transcoder
+      // Keeps application/x-infinispan-marshalling available for backwards compatibility
       encoderRegistry.registerTranscoder(new TranscoderMarshallerAdapter(globalMarshaller.wired()));
```

Comment:
This will only be possible whilst the GlobalMarshaller still exists in it's current format. Once the protostream global marshaller finally lands (12 :crossed_fingers:) we won't have the externalizers or the marshaller code.

Reply:
IDK, I only deprecated `application/x-infinispan-marshalling` in this PR, so we may have to keep the externalizers for a while longer

Reply:
In order to maintain `application/x-infinispan-marshalling` bytes compatiblity we would also have to keep the current `GlobalMarshaller` implementation, as well as the externalizers. As we haven't provided guarantess for wire compatibility, I think this has an unacceptable maintenance cost.

Reply:
I changed the deprecation comment for `application/x-infinispan-marshalling` to remove the 14.0 reference and say it'll be removed together with `AdvancedExternalizer` instead. I think this line needs to stay as is for now, though.